aiida/orm/nodes/data/array/bands.py

```diff
@@ -825,7 +825,10 @@ def _prepare_mpl_singlefile(self, *args, **kwargs):
 
         s_header = matplotlib_header_template.substitute()
         s_import = matplotlib_import_data_inline_template.substitute(all_data_json=json.dumps(all_data, indent=2))
-        s_body = matplotlib_body_template.substitute()
+        if len(all_data['paths']) == 1:
```

Comment:
Since this logic is always the same and you are repeating yourself, perhaps define a 
```
def _get_mpl_body_template(paths):
 ...
```